ecs/context.go

```diff
@@ -44,60 +43,67 @@ func newContextCreateHelper() contextCreateAWSHelper {
 	}
 }
 
-func (h contextCreateAWSHelper) createContextData(_ context.Context, opts ContextParams) (interface{}, string, error) {
-	accessKey := opts.AwsID
-	secretKey := opts.AwsSecret
+func (h contextCreateAWSHelper) createProfile(name string) error {
+	accessKey, secretKey, err := h.askCredentials()
+	if err != nil {
+		return err
+	}
+	if accessKey != "" && secretKey != "" {
+		return h.saveCredentials(name, accessKey, secretKey)
+	}
+	return nil
+}
 
+func (h contextCreateAWSHelper) createContextData(_ context.Context, opts ContextParams) (interface{}, string, error) {
 	ecsCtx := store.EcsContext{
 		Profile: opts.Profile,
 		Region:  opts.Region,
 	}
-
-	if h.missingRequiredFlags(ecsCtx) {
-		profilesList, err := h.getProfiles()
-		if err != nil {
-			return nil, "", err
-		}
-		// get profile
-		_, ok := profilesList[ecsCtx.Profile]
-		if !ok {
-			profile, err := h.chooseProfile(profilesList)
-			if err != nil {
-				return nil, "", err
-			}
-			ecsCtx.Profile = profile
-		}
-		// set region
-		region, err := h.chooseRegion(ecsCtx.Region, profilesList[ecsCtx.Profile])
-		if err != nil {
-			return nil, "", err
-		}
-		ecsCtx.Region = region
-
-		accessKey, secretKey, err = h.askCredentials()
-		if err != nil {
-			return nil, "", err
+	profilesList, err := h.getProfiles()
+	if err != nil {
+		return nil, "", err
+	}
+	if ecsCtx.Profile != "" {
+		// validate profile
+		if ecsCtx.Profile != "default" && !contains(profilesList, ecsCtx.Profile) {
+			return nil, "", fmt.Errorf("profile not found")
 		}
 	}
-	if accessKey != "" && secretKey != "" {
-		if err := h.saveCredentials(ecsCtx.Profile, accessKey, secretKey); err != nil {
-			return nil, "", err
+
+	if h.haveRequiredFlags(ecsCtx) {
```

Comment:
We appear to only use this function here, could we move the logic here instead.

Reply:
I refactored a bit to avoid duplication.